flag_supported_shows.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * This function is designed to accept a list of shows
+ * and mark each one as supported or not. The order of the shows
+ * is important as it reflects how popular the shows are.
+ *
+ * The rules are as follows:
+ *
+ * - Only one of each type of show can be supported at any time
+ * - If all the shows are the same type, they are all supported
+ * - If there are more than one show for a given type, the more popular
+ *   one is the supported one.
+ *
+ * This function returns an array of the supported shows
+ *
+ * @params shows - Array of show items: { name, type }
+ */
```

Comment:
Can we add a comment to document the optional param of `howManyOfEachTypeToSupport` for `flagSupportedShows()`
```suggestion
 * @params howManyOfEachTypeToSupport - integer of how many shows of each type to return. (optional) (default = 1)
 */
```